postgres/src/config.rs

```diff
@@ -43,6 +44,19 @@ use tokio_postgres::{Error, Socket};
 ///     path to the directory containing Unix domain sockets. Otherwise, it is treated as a hostname. Multiple hosts
 ///     can be specified, separated by commas. Each host will be tried in turn when connecting. Required if connecting
 ///     with the `connect` method.
+/// * `hostaddr` - Numeric IP address of host to connect to. This should be in the standard IPv4 address format,
+///     e.g., 172.28.40.9. If your machine supports IPv6, you can also use those addresses.
+///     If this parameter is not specified, the value of `host` will be looked up to find the corresponding IP address,
+///     - or if host specifies an IP address, that value will be used directly.
+///     Using `hostaddr` allows the application to avoid a host name look-up, which might be important in applications
+///     with time constraints. However, a host name is required for verify-full SSL certificate verification.
+///     Specifically:
+///         * If `hostaddr` is specified without `host`, the value for `hostaddr` gives the server network address.
+///             The connection attempt will fail if the authentication method requires a host name;
+///         * If `host` is specified without `hostaddr`, a host name lookup occurs;
+///         * If both `host` and `hostaddr` are specified, the value for `hostaddr` gives the server network address.
+///             The value for `host` is ignored unless the authentication method requires it,
+///             in which case it will be used as the host name.
```

Comment:
this API change feels a little brittle/confusing to me and I'm wondering if we could rework it to be more straightforward somehow. It just seems awkward that there is in implicit order-based mapping between two separate parameters. Is there a reason not to change the semantics of the `host` parameter itself?

One idea would be to use `:` to specify host->IP mapping in the `host` parameter, so you could do something like this:
```
host=host1:127.0.0.1:127.0.0.2,host2:127.0.0.3,host3
```
specifying that `host1` maps to the first two IPs, and `host2` maps to the 3rd, and `host3` should be resolved itself